src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java

```diff
@@ -241,6 +225,22 @@ private CallbackResponse handleDefendAllClaim(CallbackParams callbackParams) {
             .build();
     }
 
+    private void updateDisputesOrPartAdmission(CaseData caseData, CaseData.CaseDataBuilder<?, ?> updatedCase) {
+        if (hasPaidlessThanAndDisputesClaimOrPartAdmission(caseData)) {
+            updatedCase.specDisputesOrPartAdmission(YES);
+        } else {
+            updatedCase.specDisputesOrPartAdmission(NO);
+        }
+    }
+
+    private boolean hasPaidlessThanAndDisputesClaimOrPartAdmission(CaseData caseData) {
+        return RespondentResponseTypeSpecPaidStatus.PAID_LESS_THAN_CLAIMED_AMOUNT
+            != caseData.getRespondent1ClaimResponsePaymentAdmissionForSpec()
+            && (DISPUTES_THE_CLAIM.equals(caseData.getDefenceRouteRequired())
+            || caseData.getRespondent1ClaimResponseTypeForSpec()
```

Comment:
If I'm reading correctly, this only caters for the first respondent.

Previous code was catering also for the second respondent.

Not sure if you finished working on it, but I'll leave a comment here just in case.

Reply:
Hi Filippo,
I have changed the above and few other changes. I'll still looking to make further refactoring. 
Please take a look and let me know if the above approach is fine.